### ISSUE TITLE
🚧 Dev ➾ Refactor Internal Tasks to Run Like Plugin Tasks

### ISSUE THREAD

**Issue by hu3man** [ecadlabs/taqueria#1342]
### 🚥 Definition of Done

- [x] 🎯 Planning and Design
- [ ] 🏎️ Dev Implementation (PR merged)
- [ ] ✍️ Release Notes Added
- [ ] 🏆 Product Acceptance

### ⚖️ Acceptance Criteria

✨ Functional: 
- [ ] Refactor remaining internal tasks  so that the execution flow matches that of plugin-provided tasks 

 📐 Non-Functional: 
- [ ] ... 
 

### 🔭 Scope

🙅‍♂️ Out of Scope: 
- 
 
 

### 🧰 Work Description

This ticket covers the work involved in refactoring the CLI so that internal tasks execute the same way as plugin tasks do. This will simplify the code base, fix several known bugs, and unblocks GA in VScE 

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by hu3man** [comment]
Currently blocked by Zod refactor

**Comment by jevonearth** [comment]
Resolved as per PR #1613 :tada:
